presto-tests/src/test/java/com/facebook/presto/execution/TestHistoryBasedStatsTracking.java

```diff
@@ -129,6 +130,48 @@ public void testHistoryBasedStatsCalculator()
                 anyTree(node(AggregationNode.class, node(ExchangeNode.class, anyTree(any()))).withOutputRowCount(3).withOutputSize(54)));
     }
 
+    @Test
+    public void testHistoryBasedStatsCalculatorEnforceTimeOut()
```

Comment:
this looks like it tests cases where the timeout is enabled, but never exceeded. Can you add a test case where the timeout gets enforced, eg. you set the timeout really low (like to 0?) and then check that we don't use hbo stats?

Reply:
Added test with time limit to be 0